jenkins/pipelines/cd/tidb-operator.groovy

```diff
@@ -0,0 +1,409 @@
+    }
+    stages {
+        stage("PARAMS") {
+            steps {
+                script {
+                    ReleaseTag = params.ReleaseTag
+                    if (!ReleaseTag) {
+                        ReleaseTag = params.GitRef
+                    }
+                    println("ReleaseTag: $ReleaseTag")
+                }
+            }
+        }
+        stage("BUILD") {
+            stages {
+                stage("Bin") {
+                    agent {
+                        kubernetes {
+                            yaml goBuildYaml
+                            defaultContainer 'builder'
+                            cloud "kubernetes-ng"
+                            namespace "jenkins-tidb-operator"
+                        }
+                    }
+                    stages {
+                        stage("checkout") {
+                            steps {
+                                script {
+                                    sh '''curl -O http://fileserver.pingcap.net/download/tmp/tidb-operator.tar.gz
+                                          tar -xzf tidb-operator.tar.gz
+                                          rm tidb-operator.tar.gz'''
+                                    def scmVars = checkout changelog: false, poll: false, scm: [
+                                            $class           : 'GitSCM',
+                                            branches         : [[name: "${params.GitRef}"]],
+                                            userRemoteConfigs: [[
+                                                                        refspec: '+refs/heads/*:refs/remotes/origin/* +refs/pull/*:refs/remotes/origin/pull/*',
+                                                                        url    : 'https://github.com/pingcap/tidb-operator.git',
+                                                                ]]
+                                    ]
+                                    sh '''git clean -f .'''
+                                    GitHash = scmVars.GIT_COMMIT
+                                }
+                                println("GitHash: $GitHash")
+                            }
+                        }
+                        stage("bin") {
+                            steps {
+                                sh """set -eux
+                                    go env -w GOPROXY="https://goproxy.pingcap.net,direct"
+                                    go mod download
+                                    git checkout -- go.sum
+                                    """
+                                sh "git status"
+                                sh "GOOS=linux GOARCH=arm64 make build"
+                                sh "GOOS=linux GOARCH=amd64 make build"
+                                // todo only x86
+                                sh "GOOS=linux GOARCH=amd64 make debug-build"
```

Comment:
too many tiny steps

Reply:
any ideas?  every step produce different production

Reply:
I think split may looks better, they produce different target, they are joined in one stage just for reduce time